functions/accurate_table_generator_test.cpp

```diff
@@ -0,0 +1,54 @@
+#include "functions/accurate_table_generator.hpp"
+
+#include <iomanip>
+#include <string>
+
+#include "boost/multiprecision/cpp_int.hpp"
+#include "functions/multiprecision.hpp"
+#include "glog/logging.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "testing_utilities/approximate_quantity.hpp"
+#include "testing_utilities/is_near.hpp"
+#include "testing_utilities/numerics_matchers.hpp"
+
+namespace principia {
+namespace functions {
+namespace _accurate_table_generator {
+
+using ::testing::AnyOf;
+using namespace boost::multiprecision;
+using namespace principia::functions::_multiprecision;
+using namespace principia::testing_utilities::_approximate_quantity;
+using namespace principia::testing_utilities::_is_near;
+using namespace principia::testing_utilities::_numerics_matchers;
+
+class AccurateTableGeneratorTest : public ::testing::Test {};
+
+TEST_F(AccurateTableGeneratorTest, Sin5) {
+  auto const x = ExhaustiveSearch<5>({Sin}, 5.0 / 128.0);
+  EXPECT_EQ(x, cpp_rational(2814749767106647, 72057594037927936));
+  EXPECT_THAT(static_cast<double>(x),
+              RelativeErrorFrom(5.0 / 128.0, IsNear(3.1e-14_(1))));
+
+  // The stupid language doesn't allow printing a float in binary.  So to verify
```

Comment:
It might be useful to write a utility that prints a float in binary, or just produces a range of digits; this test is not exactly readable, and we will probably want to do more of that sort of thing.

Reply:
I'll need `ToMathematica` anyway for these Boost types, so that will be a better basis for checking the output.